File: challenges/leetcode/dartlang/lib/twosum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> hm = {};
    for (var i = 0; i < nums.length; i++) {
      var c = target - nums[i];
      if (hm.containsKey(c)) {
        return [hm[c]!, i];
      }
      hm[nums[i]] = i;
    }
    return [];
  }
}

File: csharp/LeetCode/TwoSum.cs
namespace TwoSum
{


    public class Solution
    {

        public int[] TwoSum(int[] nums, int target)
        {
            var hm = new Dictionary<int, int>();

            for (int i = 0; i < nums.Length; i++)
            {
                var compl = target - nums[i];
                if (hm.ContainsKey(compl))
                {
                    int val;
                    hm.TryGetValue(compl, out val);
                    return [i, val];
                }

                hm.TryAdd(nums[i], i);
            }
            return [];
        }
    }
}

